src/buildApplication/processVercelOutput.ts

```diff
@@ -0,0 +1,192 @@
+ * against the map of functions. The path name from the request will have a leading slash.
+ *
+ * It might be possible to accidentally call a middleware function if the request path name matches
+ * the middleware path name in the functions map, so to avoid accidental calls, and to match the
+ * value in the build output config, we remove the leading slash from the key in the map for each
+ * middleware path.
+ *
+ * @param processedOutput Map of path names to function entries.
+ * @param middlewarePaths Set of middleware paths.
+ */
+function rewriteMiddlewarePaths(
+	processedOutput: Map<string, BuildOutputItem>,
+	middlewarePaths: Set<string>
+): void {
+	for (const middlewarePath of middlewarePaths) {
+		const withLeadingSlash = addLeadingSlash(middlewarePath);
+		const entry = processedOutput.get(withLeadingSlash);
+
+		if (entry?.type === 'function') {
+			processedOutput.set(middlewarePath, { ...entry, type: 'middleware' });
+			processedOutput.delete(withLeadingSlash);
+		} else {
+			cliWarn(`Middleware path '${middlewarePath}' does not have a function.`);
+		}
+	}
+}
+
+/**
+ * Apply the overrides from the Vercel build output config to the processed output map.
+ *
+ * The overrides are used to override the output of a static asset. This includes the path name it
+ * will be served from, and the content type.
+ *
+ * @example
+ * ```
+ * // Serve the static file `500.html` from the path `/500` with the content type `text/html`.
+ * { '500.html': { path: '500', contentType: 'text/html' } }
+ * ```
+ *
+ * @link https://vercel.com/docs/build-output-api/v3/configuration#overrides
+ *
+ * @param vercelConfig Processed Vercel build output config.
+ * @param vercelOutput Map of path names to function entries.
+ */
+function applyVercelOverrides(
+	{ overrides }: ProcessedVercelConfig,
+	vercelOutput: Map<string, BuildOutputItem>
+): void {
+	if (overrides) {
+		for (const [
+			rawAssetPath,
+			{ path: rawServedPath, contentType },
+		] of Object.entries(overrides)) {
+			const assetPath = addLeadingSlash(rawAssetPath);
+			const servedPath = addLeadingSlash(rawServedPath);
+
+			const newValue: BuildOutputStaticOverride = {
+				type: 'override',
+				path: assetPath,
+				contentType,
+			};
+
+			// Update the existing static record to contain the new `contentType` and `assetPath`.
+			const existingStaticRecord = vercelOutput.get(assetPath);
+			if (existingStaticRecord?.type === 'static') {
+				vercelOutput.set(assetPath, newValue);
+			}
+
+			// Add the new served path to the map, overriding the existing record if it exists.
+			if (servedPath) {
+				vercelOutput.set(servedPath, newValue);
+			}
+
+			// If the served path is an index route, add a squashed version of the path to the map.
+			const strippedServedPath = stripIndexRoute(servedPath);
+			if (strippedServedPath !== servedPath) {
+				vercelOutput.set(strippedServedPath, newValue);
+			}
+		}
+	}
```

Comment:
The `if` + `for of` is not super readable in my opinion, what about something like this?

```suggestion
        
        Object.entries(overrides ?? []).forEach(([
			rawAssetPath,
			{ path: rawServedPath, contentType },        
        ]) => {
			const assetPath = addLeadingSlash(rawAssetPath);
			const servedPath = addLeadingSlash(rawServedPath);

			const newValue: BuildOutputStaticOverride = {
				type: 'override',
				path: assetPath,
				contentType,
			};

			// Update the existing static record to contain the new `contentType` and `assetPath`.
			const existingStaticRecord = vercelOutput.get(assetPath);
			if (existingStaticRecord?.type === 'static') {
				vercelOutput.set(assetPath, newValue);
			}

			// Add the new served path to the map, overriding the existing record if it exists.
			if (servedPath) {
				vercelOutput.set(servedPath, newValue);
			}

			// If the served path is an index route, add a squashed version of the path to the map.
			const strippedServedPath = stripIndexRoute(servedPath);
			if (strippedServedPath !== servedPath) {
				vercelOutput.set(strippedServedPath, newValue);
			}
	});
```